crates/service-client/src/http.rs

```diff
@@ -182,34 +182,22 @@ impl HttpClient {
         Either::Left(async move {
             match fut.await {
                 Ok(res) => Ok(res),
-                Err(err) if is_possible_h11_only_error(&err) => {
-                    Err(HttpError::PossibleHTTP11Only(err))
-                }
-                Err(err) => Err(HttpError::Hyper(err)),
+                Err(err) => Err(err.into()),
             }
         })
     }
 }
 
-fn is_possible_h11_only_error(err: &hyper_util::client::legacy::Error) -> bool {
-    // this is the error we see from the h2 lib when the server sends back an http1.1 response
-    // to an http2 request. http2 is designed to start requests with what looks like an invalid
-    // HTTP1.1 method, so typically 1.1 servers respond with a 40x, and the h2 client sees
-    // this as an invalid frame.
-    err.source()
-        .and_then(|err| err.downcast_ref::<h2::Error>())
-        .and_then(|err| err.reason())
-        == Some(h2::Reason::FRAME_SIZE_ERROR)
-}
-
 #[derive(Debug, thiserror::Error)]
 pub enum HttpError {
-    #[error(transparent)]
-    Hyper(#[from] hyper_util::client::legacy::Error),
     #[error(transparent)]
     Http(#[from] http::Error),
-    #[error("server possibly only supports HTTP1.1, consider discovery with --use-http1.1: {0}")]
+    #[error("server possibly only supports HTTP1.1, consider discovery with --use-http1.1.\nReason: {}", FormatHyperError(.0))]
```

Comment:
Not sure about `\n` in those messages, but let's give this a try and see how what comes out of it.